src/DiIiS-NA/D3-GameServer/GSSystem/PowerSystem/Implementations/HeroSkills/Crusader.cs

```diff
@@ -302,18 +302,21 @@ public override IEnumerable<TickTimer> Main()
 
 			yield return WaitSeconds(0.2f);
 
-			var additional_targets = GetEnemiesInRadius(User.Position, ScriptFormula(3)).Actors.OrderBy(actor => PowerMath.Distance2D(actor.Position, User.Position)).Take((int)ScriptFormula(4));
+			var additionalTargets = GetEnemiesInRadius(User.Position, ScriptFormula(3)).Actors
+				.OrderBy(actor => PowerMath.Distance2D(actor.Position, User.Position))
+				.Take((int)ScriptFormula(4))
+				.ToArray();
 
-			foreach (var target in additional_targets)
+			foreach (var target in additionalTargets)
 			{
 				target.PlayEffectGroup(RuneSelect(336292, 338256, 338255, 338254, 343105, 336292));
-				AttackPayload additional_attack = new AttackPayload(this);
-				additional_attack.SetSingleTarget(target);
-				additional_attack.AddWeaponDamage(ScriptFormula(9), DamageType.Holy);
-				additional_attack.Apply();
+				AttackPayload additionalAttack = new AttackPayload(this);
+				additionalAttack.SetSingleTarget(target);
+				additionalAttack.AddWeaponDamage(ScriptFormula(9), DamageType.Holy);
+				additionalAttack.Apply();
 				if (Rune_A > 0)         //Shared fates (buff slot 0)
-					foreach (var otherTarget in additional_targets)
```

Comment:
Here we had multiple enumeration (`OrderBy`) on each iteration of the outer loop